plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergV2.java

```diff
@@ -285,6 +287,61 @@ public void testSelectivelyOptimizingLeavesEqualityDeletes()
         Assertions.assertThat(loadTable(tableName).currentSnapshot().summary().get("total-equality-deletes")).isEqualTo("1");
     }
 
+    @Test
+    public void testEqualityDeletesWithDifferentSchemas()
+            throws Exception
+    {
+        String tableName = "test_equality_delets_different_schemas_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT * FROM tpch.tiny.nation", 25);
+        Table icebergTable = updateTableToV2(tableName);
+        Assertions.assertThat(icebergTable.currentSnapshot().summary().get("total-equality-deletes")).isEqualTo("0");
+        Path metadataDir = new Path(metastoreDir.toURI());
+        TrinoFileSystem fs = HDFS_FILE_SYSTEM_FACTORY.create(SESSION);
+
+        String deleteFile1 = "delete_file_" + UUID.randomUUID();
+        List<String> firstDeleteFileColumns = ImmutableList.of("regionkey", "name");
+        Schema deleteRowSchema = icebergTable.schema().select(firstDeleteFileColumns);
+        List<Integer> equalityFieldIds = firstDeleteFileColumns.stream()
+                .map(name -> deleteRowSchema.findField(name).fieldId())
+                .collect(toImmutableList());
+        Parquet.DeleteWriteBuilder writerBuilder = Parquet.writeDeletes(new ForwardingFileIo(fs).newOutputFile(new Path(metadataDir, deleteFile1).toString()))
+                .forTable(icebergTable)
+                .rowSchema(deleteRowSchema)
+                .createWriterFunc(GenericParquetWriter::buildWriter)
+                .equalityFieldIds(equalityFieldIds)
+                .overwrite();
+        EqualityDeleteWriter<Record> writer = writerBuilder.buildEqualityWriter();
+
+        Record dataDelete = GenericRecord.create(deleteRowSchema);
+        try (Closeable ignored = writer) {
+            writer.write(dataDelete.copy(ImmutableMap.of("regionkey", 1L, "name", "BRAZIL")));
+        }
+        icebergTable.newRowDelta().addDeletes(writer.toDeleteFile()).commit();
+
+        String deleteFile2 = "delete_file_" + UUID.randomUUID();
+        List<String> secondDeleteFileColumns = ImmutableList.of("name", "regionkey");
```

Comment:
I was checking on `IcebergPageSourceProvider` to see whether we have multiple items in `equalityDeleteSets` and we don't . 
This has probably to do with how `Schema deleteSchema = TypeUtil.select(schema, fieldIds);` internally works in iceberg (fields get pruned from the schema fields one by one if not present in the provided set).


I tried though doing the second delete filter with only "regionkey" and it works as expected.

Reply:
`Schema deleteSchema = TypeUtil.select(schema, fieldIds);` is work as you guess.

Do you have any questions about this test? I did not understand.

This test is implemented when the schema order of deleted files is different, and it can be recognized normally, officially relying on `TypeUtil.select`.

Reply:
> Do you have any questions about this test? I did not understand.

My question is what are we exactly trying to test here? It is not quite obvious to me.